packages/adapter-components/src/filters/referenced_instance_names.ts

```diff
@@ -359,52 +255,82 @@ export const createReferenceIndex = (
   return referenceIndex
 }
 
+const getPathToNestUnder = <TOptions extends APIDefinitionsOptions = {}>(
+  fetchDefinitionByType: Record<string, InstanceFetchApiDefinitions<TOptions>>,
+  instance: InstanceElement,
+  parent?: InstanceElement,
+): string[] | undefined => {
+  if (parent === undefined) {
+    return undefined
+  }
+  const shouldNestUnderParent = Object.values(
+    fetchDefinitionByType[parent.elemID.typeName]?.element?.fieldCustomizations ?? {},
+  ).find(def => def.standalone?.typeName === instance.elemID.typeName)?.standalone?.nestPathUnderParent
+  return shouldNestUnderParent
+    ? [...(parent.path?.slice(2, parent.path?.length - 1) ?? []), pathNaclCase(instance.elemID.typeName)]
+    : undefined
+}
+
+const calculateInstanceNewNameAndPath = <TOptions extends APIDefinitionsOptions = {}>(
+  instance: InstanceElement,
+  defQuery: DefQuery<InstanceFetchApiDefinitions<TOptions>, string>,
+  getElemIdFunc?: ElemIdGetter,
+  customNameMappingFunctions?: NameMappingFunctionMap<ResolveCustomNameMappingOptionsType<TOptions>>,
+): { newName: string; newPath: string[] } => {
+  const parent = getFirstParent(instance)
+  const nestUnderPath = getPathToNestUnder(defQuery.getAll(), instance, parent)
```

Comment:
readability nit 
you can -
```suggestion
  const nestUnderPath = parent ? getPathToNestUnder(defQuery.getAll(), instance, parent) : undefined 
```
and then make getPathToNestUnder to always get a parent